javaslang/src/main/java/javaslang/collection/HashArrayMappedTrie.java

```diff
@@ -117,13 +117,14 @@ static int fromBitmap(int bitmap, int bit) {
 
         abstract AbstractNode<K, V> modify(int shift, int keyHashCode, K key, V value, Action action);
 
+        @SuppressWarnings("unchecked")
         @Override
-        public Option<V> get(K key) {
-            return lookup(0, Objects.hashCode(key), key);
+        public Option<V> get(Object key) {
+            return lookup(0, Objects.hashCode(key), (K) key);
```

Comment:
Yes. We can relax signatures or make unsafe cast, but not both.
I don't like this PR, but I see no other possibility